src/SourceBuild/content/eng/finish-source-only.proj

```diff
@@ -2,12 +2,15 @@
 <Project Sdk="Microsoft.Build.NoTargets">
 
   <PropertyGroup>
+    <!-- Need to set to false to calculate RepositoryCommit. -->
+    <EnableSourceControlManagerQueries>false</EnableSourceControlManagerQueries>
     <TargetFramework>$(NetCurrent)</TargetFramework>
   </PropertyGroup>
   
   <ItemGroup>
     <ProjectReference Include="$(TasksDir)Microsoft.DotNet.UnifiedBuild.Tasks\Microsoft.DotNet.UnifiedBuild.Tasks.csproj" />
     <ProjectReference Include="$(TasksDir)Microsoft.DotNet.SourceBuild.Tasks.LeakDetection\Microsoft.DotNet.SourceBuild.Tasks.LeakDetection.csproj" />
+    <ProjectReference Include="$(RepositoryEngineeringDir)publish.proj" Targets="Build;GetMergedManifestPath" OutputItemType="MergedAssetManifest" />
```

Comment:
As mentioned, make this a normal P2P so that it works well with static graph based builds:

```suggestion
    <ProjectReference Include="$(RepositoryEngineeringDir)publish.proj" />
```